frontend/src/app/page.test.tsx

```diff
@@ -0,0 +1,47 @@
+import { describe, test, expect, vi, Mock } from "vitest"
+import { render } from "@testing-library/react"
+import Home from "./page"
+import { useSession } from "next-auth/react"
+
+vi.mock("next-auth/react", () => {
+  return {
+    __esModule: true,
+    useSession: vi.fn(),
+  }
+})
+
+const useSessionMock = useSession as Mock<typeof useSession>
+
+describe("Home", () => {
+  test("should render sign in button when there is not session", () => {
+    const mockSession = {
+      expires: new Date(Date.now() + 2 * 86400).toISOString(),
```

Comment:
nitpick: I would prevent using magic numbers, here. Could we extract this into a local (or global reused, as it is used below, too) variable?